### ISSUE TITLE
missing method-ambiguity warning for varargs

### ISSUE THREAD

**Issue by stevengj** [JuliaLang/julia#9681]
From a [julia-users discussion](https://groups.google.com/d/msg/julia-users/PMI58d5Bnt0/nlm7r5em41gJ), currently Julia does not warn about method ambiguities arising from varargs functions.  For example, compare the varargs and non-varargs cases below:
```
julia> f(x::Integer...) = 1
f (generic function with 1 method)

julia> f(x::Int, y) = 2
f (generic function with 2 methods)

julia> f(3,4)
1

julia> g(x::Integer, y::Integer) = 1
g (generic function with 1 method)

julia> g(x::Int, y) = 2
Warning: New definition 
    g(Int64,Any) at none:1
is ambiguous with: 
    g(Integer,Integer) at none:1.
To fix, define 
    g(Int64,Integer)
before the new definition.
g (generic function with 2 methods)

julia> g(3,4)
1
```
It seems to me that both cases have the same ambiguity, and both should give a warning.

**Comment by timholy** [comment]
This appears to have been fixed.

**Comment by vtjnash** [comment]
I expect this was fixed by 74dc63704b3b4b9f8b667a8e68e64678d1cf7725 (there's a test I fixed that exhibited this problem)
